kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluatorSuite.scala

```diff
@@ -284,7 +284,45 @@ class DefaultExpressionEvaluatorSuite extends AnyFunSuite with ExpressionSuiteBa
     val expOutputVector3 = booleanVector(Seq[BooleanJ](true, false, true, null))
     val actOutputVector3 = evaluator(schema, coalesceEpxr3, BooleanType.BOOLEAN).eval(batch)
     checkBooleanVectors(actOutputVector3, expOutputVector3)
+  }
+
+  test("evaluate expression: coalesce (long columns)") {
+    val longCol1 = longVector(Seq(1L, null, null, 4L))
+    val longCol2 = longVector(Seq(null, 2L, null, 5L))
+    val longCol3 = longVector(Seq(100L, null, 3L, null))
+    val longSchema = new StructType()
+      .add("longCol1", LongType.LONG)
+      .add("longCol2", LongType.LONG)
+      .add("longCol3", LongType.LONG)
+    val longBatch =
+      new DefaultColumnarBatch(longCol1.getSize, longSchema, Array(longCol1, longCol2, longCol3))
+    val longCoalesceExpr = new ScalarExpression(
+      "COALESCE",
+      util.Arrays.asList(new Column("longCol1"), new Column("longCol2"), new Column("longCol3")))
+    val expLongOutput = longVector(Seq(1L, 2L, 3L, 4L))
+    val actLongOutput = evaluator(longSchema, longCoalesceExpr, LongType.LONG).eval(longBatch)
+    checkLongVectors(actLongOutput, expLongOutput)
+  }
 
+  test("evaluate expression: coalesce (string columns)") {
```

Comment:
Can you add tests for other data types?
Ideally we cover everything but that'll be very verbose. I think it would be good to at least cover:
- timestamp (logical type is different from physical type)
- Structs (nested type)